pkg/statistics/handle/storage/stats_read_writer.go

```diff
@@ -456,6 +457,47 @@ func (s *statsReadWriter) DumpHistoricalStatsBySnapshot(
 	return jsonTbl, fallbackTbls, nil
 }
 
+// DumpStatsTOJSONBySnapshotFunc dumps statistic to json and call the function for each partition statistic.
+// Notice: It might call the function `fn` with nil jsontable.
+func (s *statsReadWriter) DumpStatsToJSONBySnapshotFunc(
```

Comment:
Could you please move this change to a separate PR? This will make it easier for the optimizer team to review.